### PR TITLE
More honest reset password request response

### LINKED ISSUE #37408: UX: Forgot password form should trim user input
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are interested into the same feature.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

**Is your feature request related to a problem? Please describe.**
Frequently users complain that they never received a password reset email. They are certain that the email is not in their spam folder and that the email address or user name they entered was correct.

The logs show that in such cases there is usually additional whitespace in the input, for example " alice" or "bob@example.com ".

**Describe the solution you'd like**
Leading and trailing whitespace from the input should be trimmed and ignored like so:

- " alice" -> "alice"
- "bob@example.com " -> "bob@example.com"

**Describe alternatives you've considered**
There could be a note in the copy that it's the user's responsibility to check for whitespace... But that would probably be even more effort than just trimming in the first place.

**Additional comments**
The copy after submitting the form is misleading (actually it's a lie):

> A password reset message has been sent to the email address of this account.

I understand that for security reasons, it should not be revealed whether this accounts exists or not, but the copy should reflect that limitation. Something along the lines of "If this account exists, an email will be sent [...]".

### PR DESCRIPTION
**Pull request by joshtrichards**
* Resolves: #37408  

## Summary

(Addresses feedback in additional comments of #37408)

Makes the response provided to users requesting a password reset more honest. Also reminds users to verify the provided account name or email address. Removes redundant copy. Continues to do so without leaking information.

## TODO

- [x] Review

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by Valdnet — COMMENTED** [review]
Change "administrator" to "administration".

File: core/src/components/login/ResetPassword.vue [hunk 1]
```diff
@@ -34,9 +34,7 @@
 
 			<NcNoteCard v-if="message === 'send-success'"
 				type="success">
-				{{ t('core', 'A password reset message has been sent to the email address of this account. If you do not receive it, check your spam/junk folders or ask your local administrator for help.') }}
-				<br>
-				{{ t('core', 'If it is not there ask your local administrator.') }}
+				{{ t('core', 'If this account exists, a password reset message has been sent to the email address of this account. If you do not receive it, verify your email address and/or account name, check your spam/junk folders or ask your local administrator for help.') }}
```
Comment (szaimen) [inline core/src/components/login/ResetPassword.vue:37]:
```suggestion
				{{ t('core', 'If this account exists, a password reset message has been sent to the email address of this account. If you do not receive it, verify your email address and/or account name, check your spam/junk folders or ask your local administration for help.') }}
```

**Review by szaimen — DISMISSED** [review]

File: core/src/components/login/ResetPassword.vue (hunk 1 above)
Comment (artonge) [inline core/src/components/login/ResetPassword.vue:37]:
```suggestion
				{{ t('core', 'If this account exists, a password reset message has been sent to its email address. If you do not receive it, verify your email address and/or account name, check your spam/junk folders or ask your local administrator for help.') }}
```

**Review by artonge — APPROVED** [review]
Nitpick, but nice otherwise :+1:

**Review by Valdnet — APPROVED** [review]

**Comment by szaimen** [comment]
Hi @joshtrichards please squash, rebase, run npm ci and npm run build and commit the changes here which will allow us to merge this. Thank you!

**Review by rakekniven — APPROVED** [review]

**Review by jtrees — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**